app/api/groups/change_password.go

```diff
@@ -23,16 +25,24 @@ func (srv *Service) changePassword(w http.ResponseWriter, r *http.Request) servi
 		return apiError
 	}
 
-retry:
-	newPassword, err := GenerateGroupPassword()
-	service.MustNotBeError(err)
+	var newPassword string
+	for retryCount := 1; ; retryCount++ {
+		if retryCount > 3 {
+			return service.ErrUnexpected(errors.New("the password generator is broken"))
+		}
+
+		newPassword, err = GenerateGroupPassword()
+		service.MustNotBeError(err)
+
+		// `CREATE UNIQUE INDEX sPassword ON groups(sPassword)` must be done
+		err = srv.Store.Groups().Where("ID = ?", groupID).Updates(map[string]interface{}{"sPassword": newPassword}).Error()
+		if err != nil && strings.Contains(err.Error(), "Duplicate entry") {
+			continue
+		}
+		service.MustNotBeError(err)
 
-	// `CREATE UNIQUE INDEX sPassword ON groups(sPassword)` must be done
-	err = srv.Store.Groups().Where("ID = ?", groupID).Updates(map[string]interface{}{"sPassword": newPassword}).Error()
-	if err != nil && strings.Contains(err.Error(), "Duplicate entry") {
-		goto retry
+		break
```

Comment:
I know people who would have been mad with a `continue` or a `break`... so here having the if case actually continuing but the continuation actually breaking, except a specific case which panics ... 🤪

Would you mind replacing these 5 lines by this (equivalent but a bit saner in terms of "flow")?
```go
if err == nil {
  break
}
if !strings.Contains(err.Error(), "Duplicate entry") {
  service.MustNotBeError(err)  // or panic(err) if you prefer
}
```

Reply:
discussed in Slack